tests/unit/gateways/clients/test_wallet_service_db_client.py

```diff
@@ -123,3 +102,90 @@ def test_get_address_balance_errors(self, engine, connection):
         cursor.one.side_effect = NoResultFound('Boom!')
         with pytest.raises(RdsNotFoundError):
             client.get_address_balance(address, token)
+
+    def test_get_address_tokens(self, engine, connection):
+        cursor = [
+            self.row_from_dict('dict_value_1'),
+            self.row_from_dict('dict_value_2'),
+        ]
+        connection.execute.return_value = cursor
+
+        client = WalletServiceDBClient(engine)
+
+        address = fake.pystr()
+        limit = fake.random_int()
+        offset = fake.random_int(min=1)
+
+        # Should return the dict value of the request
+        assert client.get_address_tokens(address, limit, offset) == ['dict_value_1', 'dict_value_2']
+
+        # Should pass the expected args to execute
+        # Should be called only once if offset > 0
+        print(connection.execute.call_args_list)
+        connection.execute.assert_called_once_with(ANY, address=address, limit=limit, offset=offset)
+        # Should use the correct query for this method
+        assert connection.execute.call_args[0][0].text == address_tokens_query
+
+    def test_get_address_tokens_with_htr(self, engine, connection):
+        def cursor_func(q, **kwargs):
+            if q.text == address_has_htr_query:
+                cursor = MagicMock()
+                cursor.one_or_none.return_value = self.row_from_dict('htr_dict_value')
+                return cursor
+            else:
+                return [
+                    self.row_from_dict('dict_value_1'),
+                    self.row_from_dict('dict_value_2'),
+                ]
+        connection.execute.side_effect = cursor_func
```

Comment:
We use `side_effect` here because we need different return values depending on which query is being executed.